lib/school_house_web/controllers/live/conference_live/index.ex

```diff
@@ -0,0 +1,70 @@
+defmodule SchoolHouseWeb.ConferenceLive.Index do
+  @moduledoc false
+
+  use SchoolHouseWeb, :live_view
+
+  alias SchoolHouse.Conferences
+
+  @impl true
+  def mount(_params, _session, socket) do
+    conferences = Conferences.list()
+    filters = default_filters()
+    countries = countries()
+
+    new_socket =
+      socket
+      |> assign(:conferences, conferences)
+      |> assign(:filters, filters)
+      |> assign(:countries, countries)
+
+    {:ok, new_socket}
+  end
+
+  @impl true
+  def handle_event("filter", params, socket) do
+    filters = update_filters(params, socket.assigns.filters)
+    conferences = query_conferences(filters)
+
+    new_socket =
+      socket
+      |> assign(:conferences, conferences)
+      |> assign(:filters, filters)
+
+    {:noreply, new_socket}
+  end
+
+  defp update_filters(%{"filters" => new_filter}, filters) do
+    atomized = for {key, val} <- new_filter, into: %{}, do: {String.to_atom(key), val}
+
+    Map.merge(filters, atomized)
+  end
+
+  defp default_filters() do
+    %{online: "false", inperson: "false", country: ""}
```

Comment:
I wonder if we could simplify this page a little by changing our filters to just "online" and "country". What do you think? 

If we're filtering by location, whether we rely on `nil` or `"Online"`/`"online"`, we can't currently support an online & in person conference. So that seems to suggest our filters are either:

- All
- Online
- By country

Then we can simplify our logic and handle combining the filters (this is untested):

```elixir
defp filtered_conferences(filters) do
  filters
  |> Enum.flat_map(&conference_filters/1)
  |> Enum.uniq()
end

defp conference_filters({"country", ""}) do
  Map.values(Conferences.by_country())
end

defp conference_filters({"country", country) do
  Map.get(Conferences.by_country(), country)
end

defp conference_filters({"online", "true"}) do
  Conferences.online()
end

defp conference_filters(_) do
  []
end
```

Reply:
Annnd finally, I am glad you suggested this. I wasn't in love with handling 10 different filter cases so this makes more sense. I used this as inspiration, but made a few adjustments.

I mentioned this in a previous comment, but now that  the "in person" filter is gone, I have simplified the logic to display the union of the results of each filter rather than the intersection.